src/Storages/MergeTree/MergeTreeData.h

```diff
@@ -474,6 +474,13 @@ class MergeTreeData : public IStorage, public WithMutableContext
 
     /// Returns absolutely all parts (and snapshot of their states)
     DataPartsVector getAllDataPartsVector(DataPartStateVector * out_states = nullptr) const;
+    
+    /// Return parts number
+    size_t getDataPartsSize() const;
+
```

Comment:
In my opinion, it's better to rename `getPartsCount` to `getActivePartsCount`, then rename this method to `getAllPartsCount`.